Src/FluentAssertions/Types/TypeSelector.cs

```diff
@@ -286,6 +286,26 @@ public TypeSelector ThatSatisfy(Func<Type, bool> predicate)
         return this;
     }
 
+    /// <summary>
+    /// Filters and returns only the types which have access modifier <paramref name="accessModifier"/>.
+    /// </summary>
+    /// <param name="accessModifier">Required access modifier for types</param>
+    public TypeSelector ThatHaveAccessModifier(CSharpAccessModifier accessModifier)
+    {
+        types = types.Where(t => t.GetCSharpAccessModifier() == accessModifier).ToList();
+        return this;
+    }
+
+    /// <summary>
+    /// Filters and returns only the types which don't have access modifier <paramref name="accessModifier"/>.
+    /// </summary>
+    /// <param name="accessModifier">Unwanted access modifier for types</param>
+    public TypeSelector ThatNotHaveAccessModifier(CSharpAccessModifier accessModifier)
```

Comment:
To line up with other methods (and to make grammatical sense,) could this be `ThatDoNotHaveAccessModifier` instead?
```suggestion
    public TypeSelector ThatDoNotHaveAccessModifier(CSharpAccessModifier accessModifier)
```

Reply:
Thanks for spotting this, missed that.